### ISSUE TITLE
DataStore not syncing with the cloud.

### ISSUE THREAD

**Issue by rupam-kundu** [aws-amplify/amplify-js#8003]
I am using Admin UI to build a web application in which React is being used in the front-end. I am trying to fetch data from a couple of DynamoDB tables namely Business and Users. However, although data exists in my tables query operations are returning empty arrays for both Business and Users tables. The following is my code.

```import React, { useState, useEffect }from "react";
import Header from "./Header";
import Jumbotron from "react-bootstrap/Jumbotron";
import { DataStore } from "@aws-amplify/datastore";
import { Users, Business } from "./models";
import Button from 'react-bootstrap/Button';

const Home = () => {
  const [userData, setUserData] = useState([]);
  const [businessData, setBusinessData] = useState([]);

  useEffect(() => {
    async function startDataStore() {
      await DataStore.start();
    }
    
    try {
      startDataStore();
    } catch (error) {
      console.error(error);
    }
    
    const userSubscription = DataStore.observe(Users).subscribe((userMsg) => {
      console.log(userMsg.model, userMsg.opType, userMsg.element);
    });

    const businessSubscription = DataStore.observe(Business).subscribe((businessMsg) => {
      console.log(businessMsg.model, businessMsg.opType, businessMsg.element);
    });

    return () => {
      userSubscription.unsubscribe();
      businessSubscription.unsubscribe();
    }

  }, []);
  
async function onQueryUsers() {
  try {
    const models = await DataStore.query(Users);
    console.log(models);
    setUserData(models);
    console.log(userData);
  } catch (error) {
    console.error(error);
  }
  return userData;
}

async function onQueryBusiness() {
  try {
    const models = await DataStore.query(Business);
    console.log(models);
    setBusinessData(models);
    console.log(businessData);
  } catch (error) {
    console.error(error);
  }
  return businessData;
}

  return (
    <>
      <Header />
      <Jumbotron>
        <h1 style={{ textAlign: "center" }}>Welcome to Appointment Notes</h1>
      </Jumbotron>
      <Button onClick={onQueryUsers}>Fetch Users</Button>
      <Button onClick={onQueryBusiness}>Fetch Businesses</Button>
      <div style={{textAlign: "center"}}>
        <h2>Users</h2>
        {
          userData.map(singleUser => (
            <div key={singleUser.id}>
              <h3>{singleUser.name}</h3>
            </div>  
          ))
        }
      </div>
      <h2>Businesses</h2>
      {
        businessData.map(singleBusiness => (
          <div key={singleBusiness.id}>
            <h3>{singleBusiness.name}</h3>
          </div>
        ))
      }
    </>
  );
};

export default Home;
```

**Comment by amhinson** [comment]
Could you try installing `@aws-amplify/ui-react@latest`? It looks like it is trying to install an older version of the package.

**Comment by iartemiev** [comment]
@rupam-kundu,

We are going to close this issue since we have not heard from you. 

Please let us know if you still need support and provide current steps to reproduce if you haven't already. We can reopen the issue to investigate further.

Thanks

**Comment by reberthkss** [comment]
Hey guys. I faced similar problem, i added Datastore to my project but it is not syncing with the cloud. I can normally create and query data locally but that data not sync with the cloud. :/

**Comment by chrisbonifacio** [comment]
@reberthkss Please open a new issue and fill out the bug report form with the details of your project configuration and some code so we can better assist you.

**Comment by reberthkss** [comment]
Hello from Brazil @chrisbonifacio ! I already solved the problem. It occurred because I'm tried to implement Datastore into an already implemented graphql model. 
Enable debug mode in datastore (`Amplify.Logger.LOG_LEVEL = 'DEBUG'`) exposed some errors between Datastore Sync Engine and the Models already in the cloud.
To resolve this I commented out all my model in `schema.graphql`, pushed to the clod (`amplify push`), uncommented my models, and pushed again. After this, the problem with sync was solved.
